### PR TITLE
Feature: liquidation price

### PR DESCRIPTION
**Pull request by shmuel-web**
# liquidation price

Screen Shot 2020-12-09 at 15 57 30

Screen Shot 2020-12-09 at 15 57 38
Screen Shot 2020-12-09 at 15 58 12

### CODE REVIEW

File: src/components/LiquidationPrice.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import React, {Component} from "react";
+import Flex, {FlexItem} from "styled-flex-component";
+import ReactTooltip from "react-tooltip"
+import styled from "styled-components"
+import {device} from "../screenSizes"
+import {getLiquidationPrice} from "../lib/Actions"
+import mainStore from "../stores/main.store"
+import {observer} from "mobx-react"
```
Comment (yaronvel) [inline src/components/LiquidationPrice.js:8]: bit `O`?

File: src/components/LiquidationPrice.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
+import React, {Component} from "react";
+import Flex, {FlexItem} from "styled-flex-component";
+import ReactTooltip from "react-tooltip"
+import styled from "styled-components"
+import {device} from "../screenSizes"
+import {getLiquidationPrice} from "../lib/Actions"
```
Comment (yaronvel) [inline src/components/LiquidationPrice.js:6]: big G?

File: src/components/LiquidationPrice.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import React, {Component} from "react";
+import Flex, {FlexItem} from "styled-flex-component";
+import ReactTooltip from "react-tooltip"
+import styled from "styled-components"
+import {device} from "../screenSizes"
```
Comment (yaronvel) [inline src/components/LiquidationPrice.js:5]: captialized d?

File: src/components/LiquidationPrice.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import {getLiquidationPrice} from "../lib/Actions"
+import mainStore from "../stores/main.store"
+import {observer} from "mobx-react"
+import moment from "moment"
+
+const  chop4 = number =>  Math.floor(parseFloat(number) * 10000) / 10000
```
Comment (yaronvel) [inline src/components/LiquidationPrice.js:11]: why 2 spaces after `const`?

File: src/stores/main.store.js [hunk 5]
```diff
@@ -28,7 +33,9 @@ class MainStore {
 
     async fetchGneralDappData () {
```
Comment (yaronvel) [inline src/stores/main.store.js:34]: typo in function name

File: src/stores/main.store.js [hunk 6]
```diff
@@ -28,7 +33,9 @@ class MainStore {
...
         await this.fetchJar()
-        this.fetchTvl() // tvl requires the spot price
+        await this.fetchTvl() // tvl requires the spot price
+        this.fetchPrices()
```
Comment (yaronvel) [inline src/stores/main.store.js:37]: why no await?

File: src/stores/main.store.js [hunk 7]
```diff
@@ -58,6 +65,29 @@ class MainStore {
...
     }
+
+    async fetchPrices () {
+        try{
+            const dataPromises = [
+                axios.get('https://defiexplore.com/api/stats/globalInfo'),
+                axios.get('https://api.coinmarketcap.com/data-api/v3/topsearch/rank')
+            ]
+            let [{data: data1}, {data: data2}] = await Promise.all(dataPromises)
+            debugger
```
Comment (yaronvel) [inline src/stores/main.store.js:76]: do we want this in the PR?

File: src/stores/main.store.js [hunk 8]
```diff
@@ -58,6 +65,29 @@ class MainStore {
...
+            data1 = data1['tokenData']['ETH-A']
+            this.makerPriceFeedPrice = data1.price
+            this.makerPriceFeedPriceNextPrice = data1.futurePrice
+            this.defiexploreLastUpdate = data1.updatedAt
+            debugger
```
Comment (yaronvel) [inline src/stores/main.store.js:81]: also this?

**Review by yaronvel — CHANGES_REQUESTED** [review]

File: src/components/LiquidationPrice.js (hunk 1 above)
Comment (shmuel-web) [reply src/components/LiquidationPrice.js:8]: not sure what do you mean:)

File: src/components/LiquidationPrice.js (hunk 2 above)
Comment (shmuel-web) [reply src/components/LiquidationPrice.js:6]: it is a function not a class
Capital letters are used for class in JS (and in some other languages as well)

File: src/components/LiquidationPrice.js (hunk 3 above)
Comment (shmuel-web) [reply src/components/LiquidationPrice.js:5]: it is an object 
same as before 
Capital letters are used for Class's in JS (and in some other languages as well)

File: src/components/LiquidationPrice.js (hunk 1 above)
Comment (yaronvel) [reply src/components/LiquidationPrice.js:8]: Big `O`, like you do in other file

**Review by yaronvel — APPROVED** [review]

**Pull request merged**